### PR TITLE
feature/SOF-7472 feat: create island defect nb

### PR DESCRIPTION
**Pull request by VsevolodX**
- feat: add island builder notebook
- update: make operational and adjust

### CODE REVIEW

File: other/materials_designer/create_island_defect.ipynb [hunk 1]
```diff
@@ -0,0 +1,350 @@
+{
```
Comment (timurbazhirov) [inline other/materials_designer/create_island_defect.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #9.    def get_coordinate_condition(shape):
</pre><p>Needs to be in made.</p><p> </p><p>Add custom_condition here</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Exabyte-io/api-examples/pull/150/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (timurbazhirov) [inline other/materials_designer/create_island_defect.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #49.        vacuum_thickness=10.0,
</pre><p>These parameters should be on top</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Exabyte-io/api-examples/pull/150/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

File: other/materials_designer/create_island_defect.ipynb [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
```
Comment (timurbazhirov) [inline other/materials_designer/create_island_defect.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #13.    CYLINDER_PARAMS = {
</pre><p>PARAMS -&gt; PARAMETERS</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Exabyte-io/api-examples/pull/150/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (timurbazhirov) [inline other/materials_designer/create_island_defect.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #43.    }
</pre><p>We should add CustomCoordinateCondition</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Exabyte-io/api-examples/pull/150/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (timurbazhirov) [inline other/materials_designer/create_island_defect.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #10.        if shape == 'cylinder':
</pre><p>If/elif should be in made, not here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Exabyte-io/api-examples/pull/150/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

File: other/materials_designer/create_slab.ipynb [hunk 3]
```diff
@@ -0,0 +1,348 @@
+{
```
Comment (timurbazhirov) [inline other/materials_designer/create_slab.ipynb]: <p>This is not clear, we should refer to README to explain what happens in Pyodide and Python.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Exabyte-io/api-examples/pull/150/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (timurbazhirov) [inline other/materials_designer/create_slab.ipynb]: <p>Remove interface</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Exabyte-io/api-examples/pull/150/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

File: utils/jupyterlite.py [hunk 4]
```diff
@@ -423,3 +423,21 @@ def write_materials_to_folder(materials: List[Any], folder_path: Optional[str] =
         with open(file_path, "w") as file:
             json.dump(material.to_json(), file)
         log(f"Material '{material.name}' written to '{file_path}'", SeverityLevelEnum.INFO, force_verbose=verbose)
+
+
+def download(content: Any, filename: str):
```
Comment (timurbazhirov) [inline utils/jupyterlite.py:428]: download_content_to_file

File: other/materials_designer/create_island_defect.ipynb (hunk 2 above)
Comment (VsevolodX) [reply other/materials_designer/create_island_defect.ipynb:1]: Added to an additional NB

File: other/materials_designer/create_slab.ipynb [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
```
Comment (timurbazhirov) [inline other/materials_designer/create_slab.ipynb:1]: <p>Update the header to match other notebook</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Exabyte-io/api-examples/pull/150/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (timurbazhirov) [inline other/materials_designer/create_slab.ipynb:1]: <p>Remove Materials are loaded according to&nbsp;<a href="https://github.com/Exabyte-io/api-examples/raw/b4c0ab1decf1490bc3a7b1297e389500fca6a166/README.ipynb" target="_blank">README</a></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Exabyte-io/api-examples/pull/150/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

File: other/materials_designer/specific_examples/defect_creation_island.ipynb [hunk 6]
```diff
@@ -0,0 +1,303 @@
+{
```
Comment (timurbazhirov) [inline other/materials_designer/specific_examples/defect_creation_island.ipynb:1]: <p>Remove "Below is the figure from the paper that we will reproduce in this notebook."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Exabyte-io/api-examples/pull/150/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

File: other/materials_designer/specific_examples/uploads/TiN.json [hunk 7]
```diff
@@ -0,0 +1 @@
...a":4.241247,"b":4.241247,"c":4.241247,"alpha":90,"beta":90,"gamma":90,"units":{"length":"angstrom","angle":"degree"},"type":"TRI","vectors":{"a":[4.241247,0,0],"b":[0,4.241247,0],"c":[0,0,4.241247],"alat":1,"units":"angstrom"}},"isNonPeriodic":false,"src":{"extension":"poscar","filename":"TiN.poscar","text":"Ti4 N4\n1.0\n   4.2412469017953978    0.0000000000000000    0.0000000000000003\n   0.0000000000000007    4.2412469017953978    0.0000000000000003\n   0.0000000000000000    0.0000000000000000    4.2412469017953978\nTi N\n4 4\ndirect\n   0.0000000000000000    0.0000000000000000    0.0000000000000000 Ti3+\n   0.0000000000000000    0.5000000000000000    0.5000000000000000 Ti3+\n   0.5000000000000000    0.0000000000000000    0.5000000000000000 Ti3+\n   0.5000000000000000    0.5000000000000000    0.0000000000000000 Ti3+\n   0.0000000000000000    0.5000000000000000    0.0000000000000000 N3-\n   0.0000000000000000    0.0000000000000000    0.5000000000000000 N3-\n   0.5000000000000000    0.5000000000000000    0.5000000000000000 N3-\n   0.5000000000000000    0.0000000000000000    0.0000000000000000 N3-\n","hash":"19c4057123b414b6b7061789519c8946"},"_id":"","metadata":{},"isUpdated":true}
```
Comment (timurbazhirov) [inline other/materials_designer/specific_examples/uploads/TiN.json:1]: name: "TiN (mp-492)"

**Review by timurbazhirov — APPROVED** [review]

File: other/materials_designer/create_slab.ipynb [hunk 8]
```diff
@@ -0,0 +1,363 @@
...
```
Comment (timurbazhirov) [inline other/materials_designer/create_slab.ipynb:1]: <p>Remove extra newline on line 1</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Exabyte-io/api-examples/pull/150/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**